### PR TITLE
fixed bug removing trailing spaces

### PR DESCRIPTION
**Pull request by hahanein**

### CODE REVIEW

File: test/csv/csv_preserve_trailing_spaces.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+    "name": "Get existing CSV file with trailing spaces in `string,array`",
+    "request": {
+        "server_url": "{{ datastore "req_base_url" }}",
+        "endpoint": "bounce-json",
+        "method": "POST",
+        "body": {{ file_csv "../_res/assets/trailing_spaces_in_string_array.csv" ',' | marshal }}
+    },
```
Comment (martinrode) [inline test/csv/csv_preserve_trailing_spaces.json:8]: all files for the test should reside in the test directory, if not shared with other tests.

File: test/csv/csv_preserve_trailing_spaces.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    "response": {
+        "statuscode": 200,
+        "body": {
+            "body": [{"id":0,"notes":["No surrounding","spaces"]},{"id":1,"notes":[" Leading","space"]},{"id":2,"notes":["Trailing ","space"]},{"id":3,"notes":[" Sorrounding ","spaces"]}]
```
Comment (martinrode) [inline test/csv/csv_preserve_trailing_spaces.json:12]: You can use a formatted json here for better readability. the apitest tool tests this as json not as string

File: pkg/lib/csv/csv_test.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
+package csv
+
+import (
+	"testing"
+)
```
Comment (martinrode) [inline pkg/lib/csv/csv_test.go:5]: for testing its better to use "github.com/stretchr/testify/assert" for asserting stuff

File: pkg/lib/csv/csv_test.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+		},
+		map[string]interface{}{
+			"id":      1,
+			"notes": []string{" Leading", "space"},
+		},
+		map[string]interface{}{
+			"id":      2,
+			"notes": []string{" Surrounding ", "space"},
+		},
+	)
```
Comment (martinrode) [inline pkg/lib/csv/csv_test.go:35]: i think its better and easier to maintain if you simply set your expectation in .json 

```
expect := `
{
   "..."
}
`

File: pkg/lib/csv/csv_test.go [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+	for i := range have {
+		for j := range have[i]["notes"].([]string) {
```
Comment (martinrode) [inline pkg/lib/csv/csv_test.go:38]: whats that? use a normal for loop

```
for _, row := range have {
...
}
```

Comment (hahanein) [reply pkg/lib/csv/csv_test.go]: Test required `row` of both `have` and `want` so went with `index`

File: pkg/lib/csv/csv_test.go (hunk 4 above)
Comment (hahanein) [reply pkg/lib/csv/csv_test.go]: So json marshal `got` for comparison?

File: pkg/lib/csv/csv_test.go (hunk 3 above)
Comment (hahanein) [reply pkg/lib/csv/csv_test.go]: It's not in your `go.mod` but your testing utils are sufficient (`AssertMapsEqual` doesn't support maps of slices though)

**Comment by martinrode** [comment]
Danke nochmal, ich habe die Änderung manuell übernommen.

**Pull request closed without merging**